File: src/main/java/com/manson/fo76/web/api/UsersController.java
package com.manson.fo76.web.api;

import com.manson.fo76.domain.User;
import com.manson.fo76.service.ItemService;
import com.manson.fo76.service.UserService;
import java.util.List;
import java.util.Objects;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping(value = "users")
public class UsersController {

	private final UserService userService;
	private final ItemService itemService;

	@Autowired
	public UsersController(UserService userService, ItemService itemService) {
		this.userService = userService;
		this.itemService = itemService;
	}

	@PostMapping(
			value = "/create", consumes = "application/json", produces = "application/json")
	public User createNewUser(@RequestBody User user) {
		return userService.createNewUser(user);
	}

	@GetMapping("/findAll")
	public List<User> findAll() {
		return userService.findAll();
	}

	@GetMapping("/findByIdOrName")
	public User findByIdOrName(String idOrName) {
		return userService.findByIdOrName(idOrName);
	}

	@DeleteMapping("/delete")
	public boolean delete(User user) {
		User userInDb = userService.delete(user);
		if (Objects.nonNull(userInDb)) {
			return itemService.deleteAllByOwnerId(userInDb.getId());
		}
		return false;
	}
}

File: src/main/java/com/manson/fo76/web/api/UsersController.kt
package com.manson.fo76.web.api

import org.springframework.web.bind.annotation.*
import springfox.documentation.annotations.ApiIgnore

@RestController
@CrossOrigin
@ApiIgnore
@RequestMapping(value = ["/users"])
class UsersController {
//@Autowired constructor(private val authenticationManager: AuthenticationManager, private val jwtTokenUtil: JwtTokenUtil,
//                                             private val userService: UserService, private val itemService: ItemService) {
//    @PostMapping(value = ["/create"], consumes = ["application/json"], produces = ["application/json"])
//    fun createNewUser(@RequestBody user: User): User? {
//        return userService.createNewUser(user)
//    }
//
//    @PostMapping(value = ["/forgot"], consumes = ["application/json"], produces = ["application/json"])
//    fun forgot(@RequestBody user: User): User? {
//        return userService.findByIdOrName(user)
//    }
//
//    @PostMapping(value = ["/token"], consumes = ["application/json"], produces = ["application/json"])
//    fun token(@RequestBody token: String?): User? {
//        val username = jwtTokenUtil.getUsernameFromToken(token)
//        return userService.findByIdOrName(username)
//    }
//
//    @GetMapping(value = ["/findAll"], produces = ["application/json"])
//    fun findAll(): List<User?>? {
//        return userService.findAll()
//    }
//
//    @GetMapping(value = ["/findByIdOrName"], produces = ["application/json"])
//    fun findByIdOrName(idOrName: String?): User? {
//        return userService.findByIdOrName(idOrName)
//    }
//
//    @DeleteMapping(value = ["/delete"], consumes = ["application/json"], produces = ["application/json"])
//    fun delete(@RequestBody user: User): Boolean {
//        val userInDb = userService.delete(user)
//        return if (Objects.nonNull(userInDb)) {
//            itemService.deleteAllByOwnerId(userInDb!!.id)
//        } else false
//    }
//
//    @PostMapping(value = ["/authenticate"], consumes = ["application/json"], produces = ["application/json"])
//    fun generateAuthenticationToken(@RequestBody user: User): ResponseEntity<*> {
//        val userDetails = userService.loadUser(user)
//                ?: return ResponseEntity.status(HttpStatus.FORBIDDEN).build<Any>()
//        authenticate(user.name, user.password)
//        val token = jwtTokenUtil.generateToken(userDetails)
//        return ResponseEntity.ok(JwtResponse(token))
//    }
//
//    private fun authenticate(username: String?, password: String?) {
//        try {
//            authenticationManager.authenticate(UsernamePasswordAuthenticationToken(username, password))
//        } catch (e: Exception) {
//            e.printStackTrace()
//        }
//    }
}